lib/transforms/serp-data-layer.js

```diff
@@ -0,0 +1,34 @@
+module.exports = function (data) {
+  var result = {};
+
+  addToLayer('query', data.q);
+  addToLayer('type', data.type);
+  addToLayer('category', data.categoryFilters);
+  addToLayer('onDisplay', data.museumFilters);
+  addToLayer('image', data.imageFilters);
+  addToLayer('archive', data.archiveFilters);
+
+  addToLayer('objectType', findSelected(data.selectedFilters.object_type));
+  addToLayer('maker', findSelected(data.selectedFilters.makers));
+  addToLayer('place', findSelected(data.selectedFilters.places));
+  addToLayer('dateFrom', findSelected(data.selectedFilters['date[from]']));
+  addToLayer('dateTo', findSelected(data.selectedFilters['date[to]']));
+
+  return JSON.stringify(result);
+
+  function addToLayer (key, val) {
```

Comment:
I think I would define this function outside the scope but I understand this implies adding the result object as a parameter. I think it would be a bit more readable.